File: 04.Simulation模拟/leetcode_2540_getCommon.cpp
#include <iostream>
#include <climits>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> cnt1, cnt2;
        int res = INT_MAX;
        for (int i : nums1) cnt1[i]++;
        for (int i : nums2) cnt2[i]++;
        for (auto& kv : cnt1){
            if (cnt2[kv.first]){
                res = min(res, kv.first);
            }
        }
        return res==INT_MAX ? -1 : res;
    }
};

int main(){
    Solution s;
    vector<int> nums1 = {1,2,3}, nums2 = {2,4};
    cout << s.getCommon(nums1, nums2) << endl;
}

File: 04.Simulation模拟/leetcode_2540_getCommon.py
from typing import List

class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        res = set(nums1).intersection(set(nums2))
        return min(res) if len(res) else -1

if __name__ == "__main__":
    s = Solution()
    nums1, nums2 = [1,2,3],[2,4]
    print(s.getCommon(nums1, nums2))
